modules/040-node-manager/templates/node-group/_bootstrap.tpl

```diff
@@ -41,13 +41,19 @@ import json
 import ssl
 
 try:
-    from urllib.request import urlopen, Request
+    from urllib.request import urlopen, Request, HTTPError
 except ImportError as e:
-    from urllib2 import urlopen, Request
+    from urllib2 import urlopen, Request, HTTPError
 
 ssl.match_hostname = lambda cert, hostname: True
 request = Request(sys.argv[1], headers={'Authorization': 'Bearer ' + sys.argv[2]})
-response = urlopen(request, cafile='/var/lib/bashible/ca.crt')
+try:
+    response = urlopen(request, cafile='/var/lib/bashible/ca.crt')
+except HTTPError as e:
+    if e.getcode() >= 401:
+        sys.stderr.write("Bootstrap-token compiled in this bootstrap.sh script is expired. Looks like more than 4 hours passed from the time it's been issued.\n")
+        sys.exit(2)
+    sys.exit(1)
```

Comment:
in another cases let re-raise exception.

Reply:
Done